src/benchmarks/gc/src/exec/GCPerfSim/GCPerfSim.cs

```diff
@@ -1725,11 +1845,13 @@ struct OldArr
 {
     public readonly ITypeWithPayload?[] items;
     public ulong TotalLiveBytes { get; private set; }
+    public uint NonEmptyLength;
 
     public OldArr(ulong numElements)
     {
         items = new ITypeWithPayload?[numElements];
         TotalLiveBytes = Util.ArraySize(items);
+        NonEmptyLength = (uint)numElements;
```

Comment:
>NonEmptyLength  [](start = 8, length = 15)

I don't understand why this is called NonEmptyLength

Reply:
In the `allocType=reference` case, we might delete some nodes from a list so that the list becomes empty. To make it easy to find a victim node to delete, these empty lists are swapped on the right (i.e. larger index) so that we can have a single number to determine where do the lists end. Any lists with an index less than this number are non-empty, and any slot greater than or equal to this number is guaranteed to be `null`. This is why it is called `NonEmptyLength`.

Reply:
makes sense.

weird, these comments simply don't show up in codeflow.